azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -421,9 +426,19 @@ def _log_error_if_none(attr_name, value):
         if ext_handler.properties.state in (None, ""):
             raise ExtensionConfigError("Received empty Extensions.Plugins.Plugin.state, failing Handler")
 
-        location = getattrib(plugin, "location")
-        failover_location = getattrib(plugin, "failoverlocation")
-        for uri in [location, failover_location]:
+        def getattrib_wrapped_in_list(node, attr_name):
+            attr = getattrib(node, attr_name)
+            return [attr] if attr else []
```

Comment:
is "if attr" a comparison against None? can we make it explicit?

Reply:
In the case of `node` being non-None but not containing an attribute named `attr_name`, getattrib returns an empty string (via `common/utils/textutil.py#91`).

Still though, I agree with the idea of making this explicit: `if attr not in (None, "")`.